app/controllers/event/affiliations_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class Event
+  class AffiliationsController < ApplicationController
+    include SetEvent
+
+    before_action :set_event, only: :create
+    before_action :set_metadata, only: [:create, :update]
+
+    def create
+      authorize @event, policy_class: AffiliationPolicy
+
+      affiliation = @event.affiliations.create(
+        {
+          name: params[:type],
+          metadata: @metadata
+        }
+      )
+
+      unless affiliation.persisted?
```

Comment:
```suggestion
      affiliation = @event.affiliations.build(
        {
          name: params[:type],
          metadata: @metadata
        }
      )

      unless affiliation.save
```
can you please error check this but this is more standard

Reply:
testing.

Reply:
it works. ty!